components/molecule/phoneInput/src/index.js

```diff
@@ -82,15 +85,27 @@ export default function MoleculePhoneInput({
           </AtomIcon>
         </div>
         <div className={`${baseClass}-input-phoneContainer`}>
-          <p className={`${baseClass}-input-prefix-code`}>
-            {selectedPrefix.countryCode}
-          </p>
+          {selectedPrefix && (
+            <p className={`${baseClass}-input-prefix-code`}>
+              {selectedPrefix.countryCode}
+            </p>
+          )}
           <AtomInput
             value={value.toString()}
-            mask={selectedPrefix.mask || '000 000 000'}
+            mask={
+              selectedPrefix && {
+                mask: isLandLine
+                  ? selectedPrefix.mask.landlineMask
+                  : selectedPrefix.mask.mobileMask
+              }
+            }
             placeholder={placeholder}
             type={TYPES.MASK}
             onChange={(e, {value}) => {
```

Comment:
In order to have clean code and be more readable, what do you think if instead of doing this `onChange` callback like this, create a `handleInputChange` method?

Reply:
Great idea!